airflow/providers/databricks/operators/databricks_sql.py

```diff
@@ -114,16 +112,17 @@ def __init__(
             **client_parameters,
             **hook_params,
         }
+        self.schema = schema
 
     def get_db_hook(self) -> DatabricksSqlHook:
         return DatabricksSqlHook(self.databricks_conn_id, **self.hook_params)
 
-    def _process_output(self, schema, results):
```

Comment:
Here `schema` was a schema of the resulting data, not what we have in `self.schema` that is the name of the schema/database...  That schema was used to write the files, and it looks like this functionality was broken by earlier refactoring.  But it looks like this functionality isn't used because we didn't receive complains. I need to think how to re-implement it.

Reply:
Just to clarity. `schema` in class constructor is the string with the name of the database, while the `schema` in parameters is the actual table schema that was returned as a query result.

Reply:
Yeah. I saw exactly this problem with schema. I think it is not a problem (other than mixture of terms - the problem is that curently it would not work - we cannot use _process_output as is becuase it expects schema as first parameter and it would get something very different. And I am quite ok with setting the schema and other fields (i have a change in progress but was a bit distracted). 

But I think we also have to fix _process_output.

The way how _process_output would work after the #25717 was rather cumbersome as I understand it:

* If the result was scalar - it would just run _process_output with FIRST column value from the row as schema, and passing the rest as extra params - if the row was exactly 2 values it would work, but otherwise it would fail with TypeError: _process_output() takes 2 positional arguments but N were given

* if the result was not scalar - it would repeat _process_output call as many rows were returned

I am working on a fix

Reply:
No. It's even more strange - Databricks Hook simply doe snot conform to the semantics of other DBApiHooks and the results are list of tuples ???? hmm.. This is quite messed up now.

Reply:
@potiuk can we split this into two fixes:

- fix for the templatization
- fix for the `process_output`
?